kubernetes/e2e_test/test_client.py

```diff
@@ -230,10 +230,6 @@ def test_exit_code(self):
         resp = api.delete_namespaced_pod(name=name, body={},
                                          namespace='default')
 
-    # Skipping this test as this flakes a lot
-    # See: https://github.com/kubernetes-client/python/issues/1300
-    # Re-enable the test once the flakiness is investigated
-    @unittest.skip("skipping due to extreme flakiness")
```

Comment:
i have not read the test carefully, but figured it's better to just ask: is there a bug in the this test you're fixing, if there is, could you please describe the bug in the test a bit?

Reply:
From the PR description: #1300

Reply:
The "bug" was that SSL sockets can have a "pending" state where data is available, but the select.select method return that the socket has data.

Reply:
I get the "bug" you're fixing in the code [ws_client.py], the fix looks good.

The question is about the changes you're making in this test file, are these changes required to make the test PASS?

Reply:
@yliaog , I am sorry, now I see what you are asking.

The test as I originally wrote it was poorly implemented, especially the loops obtaining data. They made assumptions about data arriving within 1 second. I reworked the socket select logic to have an overall timeout value and to not make assumptions about what is the maximum amount of time data will arrive on the sockets.

I suspect that such assumptions could have been contributing to the flakey'ness of the test, but I am not 100% sure.